mailchimp/mailchimp-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mailChimp/MailChimpAuthenticator.java

```diff
@@ -0,0 +1,306 @@
+            String url = this.getUserInfoEndpoint(token, authenticatorProperties);
+            String json = sendRequest(url, e);
+            JSONObject obj = new JSONObject(json);
+            if (StringUtils.isBlank(json)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Unable to fetch user claims. Proceeding without user claims");
+                }
+
+                return claims;
+            }
+
+            Map jsonObject = JSONUtils.parseJSON(json);
+            Iterator i$ = jsonObject.entrySet().iterator();
+
+            while (i$.hasNext()) {
+                Map.Entry data = (Map.Entry) i$.next();
+                String key = (String) data.getKey();
+                claims.put(ClaimMapping.build(key, key, (String) null, false), jsonObject.get(key).toString());
+                if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable("UserClaims")) {
+                    log.debug("Adding claims from end-point data mapping : " + key + " - " + jsonObject.get(key).toString());
+                }
+            }
+        } catch (Exception var11) {
+            log.error("Error occurred while accessing user info endpoint", var11);
+        }
+
+        return claims;
+    }
+
+    protected String sendRequest(String url, String accessToken) throws IOException {
+        if (log.isDebugEnabled()) {
+            log.debug("Claim URL: " + url);
+        }
+
+        if (url == null) {
+            return "";
+        } else {
+            URL obj = new URL(url);
+            HttpURLConnection urlConnection = (HttpURLConnection) obj.openConnection();
+            urlConnection.setRequestMethod("POST");
+            HttpClient httpClient = HttpClientBuilder.create().build();
+            HttpPost p = new HttpPost(url);
+
+            p.setEntity(new StringEntity("{\"apikey\":\"" + accessToken + "\"" + "}"));
+
+            HttpResponse r = httpClient.execute(p);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(r.getEntity().getContent()));
+            StringBuilder builder = new StringBuilder();
+
+            for (String inputLine = reader.readLine(); inputLine != null; inputLine = reader.readLine()) {
+                builder.append(inputLine).append("\n");
+            }
+
+            reader.close();
+            if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable("UserIdToken")) {
+                log.debug("response: " + builder.toString());
+            }
+            return builder.toString();
+        }
+    }
+
+    private OAuthClientResponse getOauthResponse(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
+            throws AuthenticationFailedException {
+        OAuthClientResponse oAuthResponse = null;
+        try {
+            oAuthResponse = oAuthClient.accessToken(accessRequest);
+        } catch (OAuthSystemException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception while requesting access token", e);
```

Comment:
Fixed in https://github.com/wso2-attic/is-connectors/pull/69/commits/d3d7a2a3e7c1ecb8309dc7a10c113b28e5c3df11